### PR TITLE
Final typing: `interpretation.py` and `helpers.py`

### PR DESCRIPTION
**Pull request by abidlabs**
This adds typing to `interpretation.py` and moves some of the interpretation-related helper methods from components to `interpretation.py`. This is the final typing-related PR so:

Closes: #2832
Closes: #2831
Closes: #2872 (not really needed)

### CODE REVIEW

**Comment by abidlabs** [comment]
Jk need to add typing to `helpers.py` too, will do that later today

**Review by aliabid94 — APPROVED** [review]
LGTM

**Comment by abidlabs** [comment]
Ok just put the final typing touches to `helpers.py`. If you wanna take another look @aliabid94, that would be great

File: gradio/helpers.py [hunk 1]
```diff
@@ -433,7 +438,8 @@ def __call__(
             total: estimated total number of steps.
             unit: unit of iterations.
         """
-        if self._active:
+        if self._callback:
+            assert self._callback is not None, "Callback not set."
```
Comment (aliabid94) [inline gradio/helpers.py:442]: this assertion can never be reached?

**Review by aliabid94 — APPROVED** [review]
lgtm

File: gradio/helpers.py (hunk 1 above)
Comment (abidlabs) [reply gradio/helpers.py:442]: Oops meant to remove that, thanks for the catch

**Pull request merged**